chunky/src/java/se/llbit/chunky/renderer/RenderManager.java

```diff
@@ -18,443 +19,128 @@
-    this.snapshotControl = callback;
-  }
+  /**
+   * Set the render taskTracker.
+   */
+  void setRenderTask(TaskTracker.Task task);
 
-  @Override public void setRenderTask(TaskTracker.Task task) {
-    renderTask = task;
-  }
+  /**
+   * Add a listener that is called on every frame.
+   */
+  void addRenderListener(RenderStatusListener listener);
+  void removeRenderListener(RenderStatusListener listener);
 
   /**
-   * Set CPU load percentage.
-   *
-   * @param value new load percentage.
+   * Run something with the buffered image (unsynchronized).
    */
-  @Override public void setCPULoad(int value) {
-    cpuLoad = value;
-  }
+  void withBufferedImage(Consumer<BitmapImage> bitmap);
 
   /**
-   * Stop render workers.
+   * Add a listener for the scene status tooltip.
+   * This is called after every preview render.
    */
-  private synchronized void stopWorkers() {
-    for (Thread worker : workers) {
-      worker.interrupt();
-    }
-  }
+  void addSceneStatusListener(SceneStatusListener listener);
+  void removeSceneStatusListener(SceneStatusListener listener);
 
-  @Override public synchronized void addSceneStatusListener(SceneStatusListener listener) {
-    sceneListeners.add(listener);
-  }
+  RenderStatus getRenderStatus();
 
-  @Override public synchronized void removeSceneStatusListener(SceneStatusListener listener) {
-    sceneListeners.remove(listener);
-  }
+  /**
+   * Start up the renderer.
+   * This should start all worker threads used by the renderer.
+   */
+  void start();
 
-  @Override public RenderStatus getRenderStatus() {
-    RenderStatus status;
-    synchronized (bufferedScene) {
-      status = new RenderStatus(bufferedScene.renderTime, bufferedScene.spp);
-    }
-    return status;
-  }
+  /**
+   * Wait for the renderer to terminate. This should only be done
+   * in headless rendering, otherwise the renderer will not automatically
+   * shut down after the render completes.
+   */
+  void join() throws InterruptedException;
 
-  @Override public void withSampleBufferProtected(SampleBufferConsumer consumer) {
-    // Synchronizing on bufferedScene ensures that we are outside the frame rendering loop.
-    synchronized (bufferedScene) {
-      consumer.accept(bufferedScene.getSampleBuffer(), bufferedScene.width, bufferedScene.height);
-    }
-  }
+  /**
+   * Run something with the sample buffer (synchronized).
+   */
+  void withSampleBufferProtected(SampleBufferConsumer consumer);
 
   /**
-   * Sends scene status text to the render preview tooltip.
+   * Shut down the renderer.
+   * This should interrupt all worker threads used by the renderer.
    */
-  private synchronized void sendSceneStatus(String status) {
-    for (SceneStatusListener listener : sceneListeners) {
-      listener.sceneStatus(status);
-    }
-  }
+  void shutdown();
 
-  @Override public void shutdown() {
-    interrupt();
+  interface SampleBufferConsumer {
+    void accept(double[] samples, int width, int height);
```

Comment:
This will be cleaner once we have an actual object for the sample buffer :partying_face: (#786)